### ISSUE TITLE
Style guidelines

### ISSUE THREAD

**Issue by piranna** [NodeOS/NodeOS#147]
Due to the discussion with @secrettriangle about the [welcome message](https://github.com/NodeOS/NodeOS/issues/141) I think we could disscuss about basic guidelines to give some uniform image of NodeOS, both on terminal or graphic UIs or merchandising :-P First points I propose:

* use cold color tones, being the primary ones (the "corporate image") gray, light blue and dark blue. This is the same as the used on the current logo and webpage. Secondary colors i think they would be good green and red, the same ones used on Node.js and npm. It would be good if the tonalities can be used on a terminal.
* basic shapes are hexagon and 60º isometric perspective (that makes cubes to degenerate on hexagons). Elements should be aligned and adjusted to an isometric grid, and from there extend to create vertical and horizontal lines (I think BeOS did this too for the file manager and icons).
* 2.5D interface, elements not so detailed as MacOS X used to have but also not a flat interface as it's being used everyway this days (Windows Metro, OSX Yosemite, iOS 7, ChromeOS Aura...) where it's dificult to identify the border of the elements. Tango icons and Elementary OS are good references.

**Comment by piranna** [comment]
For the HTML-based interface, this has some good references about how to customice the components:

https://signalvnoise.com/posts/2609-customizing-web-forms-with-css3-and-webkit

**Comment by netlovers** [comment]
The linked article is quite old, but I'm primarily a web developer, so if we get there, I can help designing a nice user interface. Although I think the newer Chrome OS look and feel is one of the nicest OS design that I've seen. It is minimalistic, but perfectly familiar, and usable.
The main thing to think about is the HTML structure of the apps so they can be super easily styled however you want. So if someone likes the Material design, they can change the main css file, and the apps look perfect, like on this OS: http://papyros.io/

**Comment by piranna** [comment]
> The linked article is quite old, but I'm primarily a web developer, so if we get there, I can help designing a nice user interface.

The link is about (between other things) show the barenaked graphic widgets interface (buttons, checkboxes...), since we'll not have a graphic library like GTK (does EGL provides some graphic widget at all?), so we'll need to draw them from scratch on the HTML renderer from scratch ;-)

Anyway, this post is mostly about to think the general styleguide, independly of being used on terminal, canvas or HTML, that's why I started proposing a "comporate color" to be used as basis for the rest of components.


> Although I think the newer Chrome OS look and feel is one of the nicest OS design that I've seen. It is minimalistic, but perfectly familiar, and usable.

I haven't used ChromeOS, but the screenshoots seems nice :-) Only point I disagree on almost all current interfaces is the "flat UI" design, that I think it makes more dificult to identify the elements.


> The main thing to think about is the HTML structure of the apps so they can be super easily styled however you want.

We could open a new issue about this... :-)

I agree on this, a change of the CSS template files (widgets, windows, desktop themes...) should be just enought, without need for using Javascript at all on them. Also, this CSS files would be used by third parties to create their own NodeOS-themed apps, or change the CSS files to mimic the OSX or Windows shells... :-)

I've found [boostrap window](http://earmbrust.github.io/bootstrap-window/), that also includes it's own window manager functionality... :-D This should be an independent project, but would be useful as basis for a NodeOS desktop, and [bootstrap CSS](http://getbootstrap.com/css/) is pretty and very common these days :-)


> So if someone likes the Material design, they can change the main css file, and the apps look perfect, like on this OS: http://papyros.io/

PapyrOS looks pretty gorgeous! :-D

**Comment by netlovers** [comment]
> The link is about (between other things) show the barenaked graphic widgets interface (buttons, checkboxes...), since we'll not have a graphic library like GTK (does EGL provides some graphic widget at all?), so we'll need to draw them from scratch on the HTML renderer from scratch ;-)

Actually this is not really true. Because browsers have their own basic element stylings (a default css file, for example [this is Chrome's](https://chromium.googlesource.com/chromium/blink/+/master/Source/core/css/html.css)), that's why you can see an input even when you haven't added any css styling to it.

**Comment by piranna** [comment]
> Actually this is not really true. Because browsers have their own basic element stylings

Yeah, but by using ```-webkit-appearance: none``` style removes it ;-) That what I wanted to show, because probably CEF will not have one and that's the (ugly) style of widgets that will be shown if we don't add our own one :-)

**Comment by piranna** [comment]
By the way, thank you for pointing to the Chrome default CSS, I was not able to find it! :-D

**Comment by netlovers** [comment]
I wonder if Servo can be used as a rendering engine?
http://blog.servo.org/2015/05/01/forward/

**Comment by piranna** [comment]
Definitely yes, any html renderer that could be used on top of EGL is a good candidate. Regarding Servo, maybe this links are useful:

https://github.com/glennw/servo-shell
https://github.com/glennw/miniservo-gtk

**Comment by piranna** [comment]
https://github.com/google/material-design-lite could be used as basis for the default CSS in combination of the Chrome one to define the aspect of the NodeOS basic widgets.

On other things, I think we could use a maximized-first tabs-based interface similar to how originally ChromeOS was, maybe taking some paning concepts of Atom and Eclipse, but not only split vertical and horizontally but also take in account tabs as a third level of organization...

**Comment by piranna** [comment]
Due to my bachelor thesis, I've upgraded the proposed HIG guidelines. Thoughs?
